packages/iTwinUI-react/src/core/Table/Table.tsx

```diff
@@ -654,6 +670,15 @@ export const Table = <
     setPageSize(pageSize);
   }, [pageSize, setPageSize]);
 
+  React.useEffect(() => {
+    if (previousFilter.current === currentFilter.current) {
+      return;
+    } else {
```

Comment:
I think you could remove `else` statement and keep next two line as just code.

Reply:
Or reverse it? Like

> if (previousFilter.current !== currentFilter.current) {
      previousFilter.current = currentFilter.current;
      onFilter?.(currentFilter.current, state, instance?.filteredRows);
    }

Reply:
This has been fixed